packages/lib-user/src/hooks/usePanoptesProjects.js

```diff
@@ -0,0 +1,47 @@
+import useSWR from 'swr'
+import { projects as panoptesProjects } from '@zooniverse/panoptes-js'
+
+const SWRoptions = {
+  revalidateIfStale: true,
+  revalidateOnMount: true,
+  revalidateOnFocus: true,
+  revalidateOnReconnect: true,
+  refreshInterval: 0
+}
+
+async function fetchProjects(id) {
+  let projectsAccumulator = []
+  
+  async function getProjects (page = 1) {
+    const query = {
+      cards: true,
+      id,
+      page,
+      page_size: 100
+    }
+    const response = await panoptesProjects.get({ query })
+    const { meta, projects } = response?.body || {}
+
+    if (projects && projects.length) {
+      projectsAccumulator = projectsAccumulator.concat(projects)
+    }
+
+    if (meta?.projects?.next_page) {
+      return getProjects(meta.projects.next_page)
+    }
```

Comment:
A safety vale is a great idea! It's a large `page_size = 100` in the request as is. I can't imagine anyone would have more than a few hundred projects? Would a return if `meta.projects.page_count > 5` make sense?

Reply:
That page_size = 100 would make an extended recursion _extremely_ unlikely to happen, nice. 👍 

[minor][nerding out] But on a purely theoretical side, the safety valve for any _recursive function_ or _while loop_ has to check the _dynamic counters_ (the values that change on each loop) for the safety to work. So checking for `if (page > 5) return 'oh no!'` or `if (projects.meta.page > 5) return 'oh dear!'` are both fine. Conversely, if we set `if (projects.meta.page_count > 5) return 'curses, alas!'`, that safety valve would trigger even on the first page/first API call, since the value of projects.meta.page_count will be consistent for any given user. Curses, alas!